### PR TITLE
AIP-84 Get Mapped Task Instance Tries

### PR DESCRIPTION
**Pull request by kandharvishnu**
related to: https://github.com/apache/airflow/issues/42370

---

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/task_instances.py
Comment (rawwar) [inline airflow/api_fastapi/core_api/routes/public/task_instances.py]:
```suggestion
```

File: airflow/api_fastapi/core_api/routes/public/task_instances.py
Comment (rawwar) [inline airflow/api_fastapi/core_api/routes/public/task_instances.py:250]: map_index will show up in the OpenAPI specifications. So, the user might send map_index here.

File: airflow/api_fastapi/core_api/routes/public/task_instances.py
Comment (rawwar) [inline airflow/api_fastapi/core_api/routes/public/task_instances.py]:
```suggestion
    return TaskInstanceHistoryCollectionResponse(
        task_instances=task_instance_select,
        total_entries=len(task_instance_select),
    )
```

Comment (kandharvishnu) [reply airflow/api_fastapi/core_api/routes/public/task_instances.py]: `TaskInstanceHistoryCollectionResponse` will take the list of `TaskInstanceHistoryResponse`. 

As a first step, I am changing the `airflow.models.taskinstance.TaskInstance` to `TaskInstanceHistoryResponse` and then passing it to `TaskInstanceHistoryCollectionResponse`

Comment (rawwar) [reply airflow/api_fastapi/core_api/routes/public/task_instances.py]: can you use `cast` like here: https://github.com/apache/airflow/blob/2390504d2424aa664a752433c6a868d7b645f72a/airflow/api_fastapi/core_api/routes/public/dag_run.py#L224

File: airflow/api_fastapi/core_api/routes/public/task_instances.py
Comment (ghost) [reply airflow/api_fastapi/core_api/routes/public/task_instances.py:250]: `get_task_instance_tries` will work for non mapped tasks. For mapped tasks which requires map_index, there is a different endpoint `get_mapped_task_instance_tries`.

File: airflow/api_fastapi/core_api/routes/public/task_instances.py
Comment (rawwar) [reply airflow/api_fastapi/core_api/routes/public/task_instances.py]: If this doesn't work. please close this conversation.

**Review by pierrejeambrun — COMMENTED** [review]
Looking good thanks :)

Should be good to merge once the base branch is updated and merged.

I'll do a final review then.

File: airflow/api_fastapi/core_api/routes/public/task_instances.py
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/task_instances.py]: Why this change ? This seems odd, what are we trying to achieve ?

**Review by pierrejeambrun — COMMENTED** [review]
Just one question

File: airflow/api_fastapi/core_api/routes/public/task_instances.py
Comment (kandharvishnu) [reply airflow/api_fastapi/core_api/routes/public/task_instances.py]: changed the default value back to `-1`

**Review by pierrejeambrun — APPROVED** [review]

**Pull request merged**
